core/chains/evm/client/errors.go

```diff
@@ -221,6 +228,10 @@ func (s *SendError) IsNonceTooLowError() bool {
 	return s.is(NonceTooLow)
 }
 
+func (s *SendError) IsNonceTooHighError() bool {
```

Comment:
You would also need to handle this error in the eth_broadcaster.go file.
For reference, see how the other retryable errors are handled: https://github.com/smartcontractkit/chainlink/blob/e1e388da99e4a3f1a4660a3411e3d3e050adbf67/core/chains/evm/txmgr/eth_broadcaster.go#L491-L501

If you don't handle this error, it will be treated as fatal.

Reply:
This is already hanled in eth_broadcaster:

```
	if sendError.IsNonceTooHighError() {
		// Nethermind specific error. Nethermind throws a NonceGap error when the tx nonce is
		// greater than current_nonce + tx_count_in_mempool, instead of keeping the tx in mempool.
		// This can happen if previous transactions haven't reached the client yet.
		// The correct thing to do is assume success for now and let the eth_confirmer retry until
		// the nonce gap gets filled by the previous transactions.
		lgr.Infow("Transaction has a nonce gap.", "err", sendError.Error())
		sendError = nil
	}
```
I assumed though that the confirmer rebroadcasts any tx, but from what I've read on the code it only bumps gas for txs and confirms receipts, so the above should be changed to this:
```
 	if sendError.IsNonceTooHighError() {
		// Nethermind specific error. Nethermind throws a NonceGap error when the tx nonce is
		// greater than current_nonce + tx_count_in_mempool, instead of keeping the tx in mempool.
		// This can happen if previous transactions haven't reached the client yet.
		// The correct thing to do is to retry until the nonce gap gets filled by the previous transactions.
		lgr.Criticalw("Transaction has a nonce gap.", "err", sendError.Error())
		return sendError, false
	}
```